src/cmd/metadata.rs

```diff
@@ -107,13 +168,29 @@ impl GenerateMetadataCommand {
             .transpose()?;
         let homepage = self.crate_metadata.homepage.clone();
         let license = contract_package.license.clone();
-        let hash = self.wasm_hash()?;
-
+        //{
+        let (dest_wasm, hash, optimization_result) =
+            if self.build_artifact != GenerateArtifacts::MetadataOnly {
+                let (wasm, hash, optimization) = self.wasm_hash()?;
+                (Some(wasm), Some(hash), Some(optimization))
+            } else {
+                (None, None, None)
```

Comment:
Since this is either all `None` or all `Some`, this indicates they should be grouped together.

Reply:
So I agree, but thing is that in the subsequent code they are each used in a different context which requires them each as an `Option`. So imho this a short way of assigning the missing ones `None` here, while wrapping the others in `Some`. Otherwise this would need to be done in the subsequent code.